File: tests/fixtures/python/JavadocConversion.py
from __future__ import annotations


class JavadocConversion:
    """Normalizes input strings for review.

    References `normalize(String)` and `normalize(String, Object)` values.
    """

    def normalize(self, value: str) -> str:
        """Normalize the supplied value.

        Args:
            value: the raw value to normalize

        Returns:
            the normalized value

        Raises:
            IllegalArgumentException: when value is invalid
        """

        return value

    def clean(self, value: str) -> str:
        """Clean a value.

        .. deprecated:: use `normalize(String)` instead.

        Args:
            value: the raw value

        Returns:
            the cleaned value
        """

        return self.normalize(value)

File: tests/fixtures/java/JavadocConversion.java
package com.example;

/**
 * Normalizes input strings for review.
 *
 * <p>References {@link #normalize(String)} and {@link #normalize(String, Object)} values.
 *
 * @since 1.0
 */
public class JavadocConversion {

    /**
     * Normalize the supplied value.
     *
     * @param value the raw value to normalize
     * @return the normalized value
     * @throws IllegalArgumentException when value is invalid
     */
    public String normalize(String value) {
        return value;
    }

    /**
     * Clean a value.
     *
     * @deprecated use {@link #normalize(String)} instead.
     * @param value the raw value
     * @return the cleaned value
     */
    public String clean(String value) {
        return normalize(value);
    }
}
